# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1295762663 28800
#      Sat Jan 22 22:04:23 2011 -0800
# Node ID 2162104d40b1fa29593a112c0499233c8dc94ff8
# Parent  2f728f9d32d488a1285c8dde911418d3db0ef3c5
Revise HACKING file and incorporate all of ROADMAP file.

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2010-01-22  Rik  <octave@nomad.inbox5.com>
+
+	* HACKING: Revise and incorporate all of file ROADMAP.
+	* ROADMAP: Deleted.
+
 2010-01-22  Rik  <octave@nomad.inbox5.com>
 
 	* README.Windows: Reference README.MinGW
 	* README.MinGW: Replace placeholder with actual instructions
 
 2011-01-22  John W. Eaton  <jwe@octave.org>
 
 	* configure.ac (AC_INIT): Version is now 3.5.0+.
diff --git a/HACKING b/HACKING
--- a/HACKING
+++ b/HACKING
@@ -37,17 +37,17 @@ Later, after synchronizing from the repo
 be sufficient.
 
 ** First checkout
 
 Obviously, if you are reading these notes, you did manage to check out
 this package from the repository.  For the record, you will find all the
 relevant information on downloading sources at:
 
-  http://savannah.gnu.org/hg/?group=octave
+  http://www.gnu.org/software/octave/download.html
 
 After checking out Octave, you will need to run the autogen.sh script:
 
   $ ./autogen.sh
 
 This script will examine the source tree and generate some Makefile
 fragments and then runs the bootstrap script.  The bootstrap script comes
 from gnulib, but is kept in the Octave source archive.  It should be
@@ -86,26 +86,100 @@ run
 
 At this point, there should be no difference between your local copy,
 and the master copy:
 
   $ hg diff
 
 should output no difference.
 
-Additional information about coding style 
+** Coding style
+
+The most important advice is to follow any conventions you detect in the
+file being edited.  In addition, Octave maintainers have written a lot
+on the subject.  See "Appendix C: Tips and Standards" and
+"Appendix D: Contributing Guidelines" in the PDF documentation.
+
+* Bugs and patches
+
+See the file BUGS in this directory for more guidance on reporting bugs
+and preparing patches.
+
+* Code layout
+
+An overview of the directory layout of Octave's source files:
+
+  doc           -- Texinfo documentation for Octave
 
-This file will eventually be extended to explain more about updating
-Octave, preparing patches, and making releases.
+  examples      -- some example files
+
+  libcruft      -- various numerical libraries (mostly Fortran)
+    amos           * bessel functions
+    blas-xtra      * wrappers for blas functions used in Octave
+    daspk          * large scale differential algebraic equation solver
+    dasrt          * differential algebraic equation solver with root finding
+    dassl          * differential-algebraic system solver
+    fftpack        * subroutines for fast fourier transforms
+    lapack-xtra    * wrappers for lapack functions used in Octave
+    misc           * miscellaneous utilities
+    odepack        * odinary differential equation solver
+    ordered-qz     * code for ordering eigenvalues for QZ factorization
+    quadpack       * subroutines for numerical integration
+    ranlib         * random number generators
+    slatec-err     * slatec error handling library
+    slatec-fn      * various special function subroutines
+
+  liboctave     -- the C++ interfaces to the numerical libraries and
+                   various OS facilities
 
-Enjoy!
+  scripts       -- functions written in the Octave language
+    audio          * play and record sound files (system dependent)
+    deprecated     * older deprecated functions
+    elfun          * elementary mathematical functions
+    @ftp           * ftp object class
+    general        * utility functions
+    geometry       * geometry algorithms
+    image          * image processing
+    help           * help subsystem functions
+    io             * input/output functions
+    linear-algebra * linear algebra stuff
+    miscellaneous  * stuff that doesn't fit anywhere else
+    optimization   * zero finders and minimizers
+    path           * functions for path manipulation
+    pkg            * the package manager
+    plot           * plotting functions
+    polynomial     * polynomial manipulation
+    set            * set manipulation
+    signal         * signal processing
+    sparse         * sparse matrix support
+    specfun        * special mathematical functions
+    special-matrix * functions for generating special types of matrices
+    startup        * initialization functions
+    statistics     * statistical functions, distributions, and tests
+    strings        * character string manipulation
+    testfun        * unit testing
+    time           * time and date functions
 
------
+  src           -- the interpreter itself
+
+  test          -- tests for the interpreter
+    test_*.m       * fixed tests for the interpreter
+    fntests.m      * script to run function tests embedded in C++ and .m files
+
 
-Copyright (C) 2009 John W. Eaton
+----
+John W. Eaton
+jwe@octave.org
+
+
+Last updated: Sat Jan 22 22:00:25 PST 2011
+
+################################################################################
+
+Copyright (C) 2009,2011 John W. Eaton
 
 This file is part of Octave.
 
 Octave is free software; you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by the
 Free Software Foundation; either version 3 of the License, or (at
 your option) any later version.
 
diff --git a/ROADMAP b/ROADMAP
deleted file mode 100644
--- a/ROADMAP
+++ /dev/null
@@ -1,64 +0,0 @@
-An overview of the directory layout of Octave's source files:
-
-  doc           -- Texinfo documentation for Octave
-
-  examples      -- some example files
-
-  libcruft      -- various numerical libraries (mostly Fortran)
-    amos           * bessel functions
-    blas-xtra      * wrappers for blas functions used in Octave
-    daspk          * large scale differential algebraic equation solver
-    dasrt          * differential algebraic equation solver with root finding
-    dassl          * differential-algebraic system solver
-    fftpack        * subroutines for fast fourier transforms
-    lapack-xtra    * wrappers for lapack functions used in Octave
-    misc           * miscellaneous utilities
-    odepack        * odinary differential equation solver
-    ordered-qz     * code for ordering eigenvalues for QZ factorization
-    quadpack       * subroutines for numerical integration
-    ranlib         * random number generators
-    slatec-err     * slatec error handling library
-    slatec-fn      * various special function subroutines
-
-  liboctave     -- the C++ interfaces to the numerical libraries and
-                   various OS facilities
-
-  scripts       -- functions written in the Octave language
-    audio          * play and record sound files (system dependent)
-    deprecated     * older deprecated functions
-    elfun          * elementary mathematical functions
-    @ftp           * ftp object class
-    general        * utility functions
-    geometry       * geometry algorithms
-    image          * image processing
-    help           * help subsystem functions
-    io             * input/output functions
-    linear-algebra * linear algebra stuff
-    miscellaneous  * stuff that doesn't fit anywhere else
-    optimization   * zero finders and minimizers
-    path           * functions for path manipulation
-    pkg            * the package manager
-    plot           * plotting functions
-    polynomial     * polynomial manipulation
-    set            * set manipulation
-    signal         * signal processing
-    sparse         * sparse matrix support
-    specfun        * special mathematical functions
-    special-matrix * functions for generating special types of matrices
-    startup        * initialization functions
-    statistics     * statistical functions, distributions, and tests
-    strings        * character string manipulation
-    testfun        * unit testing
-    time           * time and date functions
-
-  src           -- the interpreter itself
-
-  test          -- tests for the interpreter
-    test_*.m       * fixed tests for the interpreter
-    fntests.m      * script to run function tests embedded in C++ and .m files
-
-
-John W. Eaton
-jwe@octave.org
-
-Last updated: Mon Jan 10 21:48:42 PST 2011
